spec/controllers/manager/users_controller_spec.rb

```diff
@@ -1,54 +1,135 @@
   let(:super_admin) { create(:super_admin) }
 
+  before { sign_in super_admin }
+
   describe '#show' do
     render_views
 
     let(:super_admin) { create(:super_admin) }
     let(:user) { create(:user) }
 
     before do
-      sign_in(super_admin)
       get :show, params: { id: user.id }
     end
 
     it { expect(response.body).to include(user.email) }
   end
 
   describe '#update' do
-    let!(:user) { create(:user, email: 'ancien.email@domaine.fr') }
+    let(:user) { create(:user, email: 'ancien.email@domaine.fr') }
 
-    before {
-      sign_in super_admin
-    }
     subject { patch :update, params: { id: user.id, user: { email: nouvel_email } } }
 
-    describe 'with a valid email' do
-      let(:nouvel_email) { 'nouvel.email@domaine.fr' }
+    context 'when the targeted email does not exist' do
+      describe 'with a valid email' do
+        let(:nouvel_email) { 'nouvel.email@domaine.fr' }
+
+        it 'updates the user email' do
+          subject
+
+          expect(User.find_by(id: user.id).email).to eq(nouvel_email)
+        end
+
+        context 'and the user is an administrateur' do
+          let(:user) { create(:administrateur).user }
 
-      it 'updates the user email' do
-        subject
+          it 'rejects the modification' do
+            subject
 
-        expect(User.find_by(id: user.id).email).to eq(nouvel_email)
+            expect(flash[:error]).to match("« nouvel.email@domaine.fr » est un administrateur. On ne sait pas encore faire.")
+            expect(user.reload.email).not_to eq(nouvel_email)
+          end
+        end
+      end
+
+      describe 'with an invalid email' do
+        let(:nouvel_email) { 'plop' }
+
+        it 'does not update the user email' do
+          subject
+
+          expect(User.find_by(id: user.id).email).not_to eq(nouvel_email)
+          expect(flash[:error]).to match("« #{nouvel_email} » n’est pas une adresse valide.")
+        end
       end
     end
 
-    describe 'with an invalid email' do
-      let(:nouvel_email) { 'plop' }
+    context 'when the targeted email exists' do
+      let(:preexisting_user) { create(:user, email: 'email.existant@domaine.fr') }
+      let(:nouvel_email) { preexisting_user.email }
+
+      context 'and the old account has a dossier' do
+        let!(:dossier) { create(:dossier, user: user) }
+
+        it 'transfers the dossier' do
+          subject
 
-      it 'does not update the user email' do
-        subject
+          expect(preexisting_user.dossiers).to match([dossier])
+        end
+      end
 
-        expect(User.find_by(id: user.id).email).not_to eq(nouvel_email)
-        expect(flash[:error]).to match("« #{nouvel_email} » n'est pas une adresse valide.")
+      context 'and the old account belongs to an instructeur and expert' do
+        let!(:instructeur) { create(:instructeur, user: user) }
+        let!(:expert) { create(:expert, user: user) }
+
+        it 'transfers instructeur account' do
+          subject
```

Comment:
Pour les `let!`, si on veut éviter, je dirais qu'on peut déclencher explicitement la création dans un `before` :

```suggestion
        before do
          instructeur
          expert
        end
```

C'est légèrement plus explicite et flexible (on peut mieux contrôler l'ordre de création), mais c'est aussi plus verbeux. Si tu veux laisser les `let!`, ici ça me semble pertinent.